lib/Substitute.js

```diff
@@ -0,0 +1,16 @@
+function substituteScope(scope) {
```

Comment:
also can you add a comment explaining the purpose of this function? I'm a bit confused about what it does - is it replacing scope things? Shouldn't that happen in the steps instead?